types/index.d.ts

```diff
@@ -1,11 +1,11 @@
 // TypeScript Version: 3.0
-import { Hook } from '@feathersjs/feathers';
+import { Hook, HookContext } from '@feathersjs/feathers';
 
 export interface PermissionOptions {
   error?: boolean;
   entity?: string;
   field?: string;
-  roles: string|string[];
+  roles: string|string[]|{(context: HookContext): string | string[]};
```

Comment:
I think it would ideally be

```js
type RoleList = string|string[];

// ...
roles: RoleList|{(context: HookContext): RoleType}|{(context: HookContext): Promise<RoleType>};